test/test-collection.js

```diff
@@ -24,25 +24,56 @@ function mockTest(opts, title) {
 }
 
 function titles(tests) {
+	if (!tests) {
+		tests = [];
+	}
+
 	return tests.map(function (test) {
 		return test.title;
 	});
 }
 
+function removeEmptyProps(obj) {
+	if (Array.isArray(obj) && obj.length === 0) {
+		return null;
+	}
+
+	if (obj.constructor !== Object) {
+		return obj;
+	}
+
+	var cleanObj;
```

Comment:
Should initialize to `null` so `removeEmptyProps({})` is consistent with `removeEmptyProps([])`.